tools/FakeItEasy.Build/Program.cs

```diff
@@ -73,6 +75,38 @@ public static void Main(string[] args)
                         }
                     });
 
+            Target(
+                "initialize-user-properties",
+                () =>
+                    {
+                        if (!File.Exists("FakeItEasy.user.props"))
+                        {
+                            var defaultUserProps = @"
+<Project>
+  <PropertyGroup>
+    <BuildProfile></BuildProfile>
+  </PropertyGroup>
+</Project>".Trim();
+                            File.WriteAllText("FakeItEasy.user.props", defaultUserProps, Encoding.UTF8);
+                        }
+                    });
+
+            foreach (var profile in Directory.EnumerateFiles("profiles", "*.props").Select(Path.GetFileNameWithoutExtension))
+            {
+                Target(
+                    "use-profile-" + profile,
+                    DependsOn("initialize-user-properties"),
+                    () =>
+                        {
+                            var originalText = File.ReadAllText("FakeItEasy.user.props", Encoding.UTF8);
+                            var newText = Regex.Replace(originalText, "(?<=<BuildProfile>)[^>]*(?=</BuildProfile>)", profile);

```

Comment:
I... don't even understand how that regex works. How about using Linq to XML? I think the code would be almost as short, and much easier to understand.

Reply:
Done. And at the risk of tooting your horn, I think it's an improvement!